### ISSUE TITLE
[wasm] ThreadPool Regression 

### ISSUE THREAD

**Issue by tqiu8** [dotnet/runtime#37818]
### Description
Threadpool works in 3.2 but not on 5.0 regardless of setting min and max threads.

### Data
With `await Task.WhenAll(tasks);` in a test

Error from a 5.0 wasm sample with ThreadPool
```
WASM-ERR: Unhandled Exception:
WASM-ERR: System.Threading.Tasks.TaskSchedulerException: An exception was thrown by a TaskScheduler.
WASM-ERR:  ---> System.PlatformNotSupportedException: Cannot start threads on this runtime.
WASM-ERR:    at System.Threading.Thread.Start()
WASM-ERR:    at System.Threading.PortableThreadPool.WorkerThread.TryCreateWorkerThread()
WASM-ERR:    at System.Threading.PortableThreadPool.WorkerThread.MaybeAddWorkingWorker()
WASM-ERR:    at System.Threading.PortableThreadPool.RequestWorker()
WASM-ERR:    at System.Threading.ThreadPool.RequestWorkerThread()
WASM-ERR:    at System.Threading.ThreadPoolWorkQueue.EnsureThreadRequested()
WASM-ERR:    at System.Threading.ThreadPoolWorkQueue.Enqueue(Object callback, Boolean forceGlobal)
WASM-ERR:    at System.Threading.ThreadPool.UnsafeQueueUserWorkItemInternal(Object callBack, Boolean preferLocal)
WASM-ERR:    at System.Threading.Tasks.ThreadPoolTaskScheduler.QueueTask(Task task)
WASM-ERR:    at System.Threading.Tasks.TaskScheduler.InternalQueueTask(Task task)
WASM-ERR:    at System.Threading.Tasks.Task.ScheduleAndStart(Boolean needsProtection)
WASM-ERR:    --- End of inner exception stack trace ---
WASM-ERR:    at System.Threading.Tasks.Task.ScheduleAndStart(Boolean needsProtection)
WASM-ERR:    at System.Threading.Tasks.Task.InternalStartNew(Task creatingTask, Delegate action, Object state, CancellationToken cancellationToken, TaskScheduler scheduler, TaskCreationOptions options, InternalTaskOptions internalOptions)
WASM-ERR:    at System.Threading.Tasks.Task.Run(Action action)
WASM-ERR:    at Test.GetSyncRootBasic()
WASM-ERR:    at Test.Main(String[] args)
WASM-ERR:    at Test.<Main>(String[] args)
Exception:System.Threading.Tasks.TaskSchedulerException: An exception was thrown by a TaskScheduler.
 ---> System.PlatformNotSupportedException: Cannot start threads on this runtime.
   at System.Threading.Thread.Start()
   at System.Threading.PortableThreadPool.WorkerThread.TryCreateWorkerThread()
   at System.Threading.PortableThreadPool.WorkerThread.MaybeAddWorkingWorker()
   at System.Threading.PortableThreadPool.RequestWorker()
   at System.Threading.ThreadPool.RequestWorkerThread()
   at System.Threading.ThreadPoolWorkQueue.EnsureThreadRequested()
   at System.Threading.ThreadPoolWorkQueue.Enqueue(Object callback, Boolean forceGlobal)
   at System.Threading.ThreadPool.UnsafeQueueUserWorkItemInternal(Object callBack, Boolean preferLocal)
   at System.Threading.Tasks.ThreadPoolTaskScheduler.QueueTask(Task task)
   at System.Threading.Tasks.TaskScheduler.InternalQueueTask(Task task)
   at System.Threading.Tasks.Task.ScheduleAndStart(Boolean needsProtection)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ScheduleAndStart(Boolean needsProtection)
   at System.Threading.Tasks.Task.InternalStartNew(Task creatingTask, Delegate action, Object state, CancellationToken cancellationToken, TaskScheduler scheduler, TaskCreationOptions options, InternalTaskOptions internalOptions)
   at System.Threading.Tasks.Task.Run(Action action)
   at Test.GetSyncRootBasic()
   at Test.Main(String[] args)
   at Test.<Main>(String[] args)
```

Results from 3.2 sample Blazor app with the same test
```
SetMinThreads returns: False 
blazor.webassembly.js:1 SetMaxThreads returns False
blazor.webassembly.js:1 
blazor.webassembly.js:1 Maximum worker threads: 	1
blazor.webassembly.js:1 Maximum completion port threads: 200
blazor.webassembly.js:1 
blazor.webassembly.js:1 Minimum worker threads: 	1
blazor.webassembly.js:1 Available completion port threads: 16
2blazor.webassembly.js:1 
blazor.webassembly.js:1 Available worker threads: 	1
blazor.webassembly.js:1 Available completion port threads: 200
blazor.webassembly.js:1 
blazor.webassembly.js:1 Completed
```

**Comment by danmosemsft** [comment]
Changing label - this fix presumably is in the runtime

**Comment by lewing** [comment]
Fixed in #38029
